client/src/game/components/GameBoard/GameBoard.tsx

```diff
@@ -132,8 +139,70 @@ export class GameBoard extends React.Component<GameBoard.Props, GameBoard.State>
     }
   }
 
-  private optimisticallyCalculateNextBoard(moveIntent: MoveIntent): number[] {
-    return this.props.gameState.board // TODO: calculate desired GameState from moveIntent
+  private optimisticallyCalculateNextBoard(moveIntent: MoveIntent, source: ChessboardLib.Square, target: ChessboardLib.Square): number[] {
+    const chess = new Chess();
+    const moveCount: number = this.props.gameState.moveCount;
+    if(moveCount === 0){
+      chess.clear();
+    }
+    else{
+      chess.load(this.getChessboardPosition());
+    }
+    
+    var board: number[] = this.props.gameState.board;
```

Comment:
we'll want to copy this using `[...this.props.gameState.board]` instead of mutating in place, which can cause bugs